packages/@webex/plugin-cc/src/services/core/WebSocket/WebSocketManager.ts

```diff
@@ -0,0 +1,166 @@
+    if (!this.isSocketClosed && this.shouldReconnect) {
+      this.shouldReconnect = shouldReconnect;
+      this.websocket.close();
+      this.keepaliveWorker.postMessage({ type: 'terminate' });
+      console.error(`[WebSocketStatus] | event=webSocketClose | WebSocket connection closed manually REASON: ${reason}`);
+    }
+  }
+
+  private async register(connectionConfig: SubscribeRequest) {
+    try {
+      const subscribeResponse: SubscribeResponse = await this.webex.request({
+        service: WCC_API_GATEWAY,
+        resource: SUBSCRIBE_API,
+        method: HTTP_METHODS.POST,
+        body: connectionConfig,
+      });
+      this.url = subscribeResponse.body.webSocketUrl;
+    } catch (e) {
+      console.error("Register API Failed", "Request to RoutingNotifs websocket registration API failed", e);
+    }
+  }
+
+  private async connect() {
+    if (!this.url) {
+      return undefined;
+    }
+    console.log(`[WebSocketStatus] | event=webSocketConnecting | Connecting to WebSocket: ${this.url}`);
+    this.websocket = new WebSocket(this.url);
+    return new Promise((resolve, reject) => {
+      this.websocket.onopen = () => {
+        this.isSocketClosed = false;
+        this.shouldReconnect = true;
+
+        this.websocket.send(JSON.stringify({ keepalive: 'true' }));
+        this.keepaliveWorker.onmessage = (keepAliveEvent: { data: any }) => {
+          if (keepAliveEvent?.data?.type === 'keepalive') {
+            this.websocket.send(JSON.stringify({ keepalive: 'true' }));
+          }
+
+          if (keepAliveEvent?.data?.type === 'closeSocket' && this.isConnectionLost) {
+            this.forceCloseWebSocketOnTimeout = true;
+            this.close(true, 'WebSocket did not auto close within 16 secs');
+            console.error('[webSocketTimeout] | event=webSocketTimeout | WebSocket connection closed forcefully');
+          }
+        };
+
+        this.keepaliveWorker.postMessage({
+          type: 'start',
+          intervalDuration: 4000, // Keepalive interval
+          isSocketClosed: this.isSocketClosed,
+          closeSocketTimeout: 16000, // Close socket timeout
+        });
+      };
+
+      this.websocket.onerror = (event: any) => {
+        console.error(`[WebSocketStatus] | event=socketConnectionFailed | WebSocket connection failed`, event);
+        reject();
+      };
+
+      this.websocket.onclose = async (event: any) => {
+        this.webSocketOnCloseHandler(event);
+      };
+
+      this.websocket.onmessage = (e: MessageEvent) => {
+        this.onMessageSend(e.data);
+        const eventData = JSON.parse(e.data);
+
+        if (eventData.type === 'Welcome') {
+          this.isWelcomeReceived = true;
+          if (this.welcomePromiseResolve) {
+            this.welcomePromiseResolve(eventData.data as WelcomeResponse);
+            this.welcomePromiseResolve = null;
+          }
+        }
+
+        if (eventData.type === 'AGENT_MULTI_LOGIN') {
```

Comment:
Again, all have been taken from agent desktop, so if we do not require this, we can remove them.
But overall the logic is pretty much the same.

Reply:
It is okay to keep this as this whole copy-paste effort is keeping the WxCC Desktop use cases in mind